extensions/pkg/webhook/registration.go

```diff
@@ -174,7 +173,7 @@ func ReconcileSeedWebhookConfig(ctx context.Context, c client.Client, webhookCon
 
 	if _, err := controllerutils.GetAndCreateOrStrategicMergePatch(ctx, c, webhookConfig, func() error {
 		if ownerReference != nil {
-			webhookConfig.SetOwnerReferences(kubernetes.MergeOwnerReferences(webhookConfig.GetOwnerReferences(), *ownerReference))
```

Comment:
I guess we need this to remove the old reference to the `garden` namespace. However, wouldn't this be a potentially breaking change for extensions in the seed?

Reply:
There can be only one owner reference of type `controller` for one object. If there is more then one, kube-apiserver rejects the update. Thus, we would not be able to change the existing owner reference if we try to merge them here.
This does not affect the extensions in the seed, because in this case the namespace for the owner reference does not change.
Afaik it does not make sense to merge owner references when the `controller` flag is set. If the reference does not change, it does nothing. If the references does change, the update will fail.